src/org/thoughtcrime/securesms/LogViewActivity.java

```diff
@@ -57,10 +58,22 @@ public boolean onPrepareOptionsMenu(Menu menu) {
   public boolean onOptionsItemSelected(MenuItem item) {
     super.onOptionsItemSelected(item);
     Float newSize;
+
     switch (item.getItemId()) {
       case android.R.id.home:
         finish();
         return true;
+      case R.id.save_log:
+        try {
+          if (logViewFragment.saveLogFile())
+            Toast.makeText(getApplicationContext(), R.string.pref_saved_log, Toast.LENGTH_LONG).show();
+          else
+            Toast.makeText(getApplicationContext(), R.string.pref_save_log_failed, Toast.LENGTH_LONG).show();
+        } catch (NoExternalStorageException e) {
```

Comment:
This try -  catch seems to be redundant (see my comments in LogViewFragment)

Reply:
Android Studio forced me to use this here (as well).

Reply:
The reason is that you've defined the method like this `.. saveLogFile() throws NoExternalStorageException`. Remove `throws NoExternalStorageException` and Android Studio will be your friend again.